src/main/java/dev/thread/MyThreadExt.java

```diff
@@ -0,0 +1,23 @@
+package dev.thread;
+
+import dev.Counter;
+import org.apache.log4j.Logger;
+
+public class MyThreadExt extends Thread {
+    private static final Logger logger
+            = Logger.getLogger(MyThreadExt.class);
+    private Counter counter;
+
+    public MyThreadExt(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        for (int i = 0; i < 100; i++) {
```

Comment:
magic number?

Reply:
make it constant and keep it in Counter. You can make there boolean method(to check if the number has been reached)  and use here while loop